cmd/keytransparency-sequencer/main.go

```diff
@@ -183,21 +167,20 @@ func main() {
 	grpc_prometheus.Register(grpcServer)
 	grpc_prometheus.EnableHandlingTimeHistogram()
 
-	glog.Infof("Signer starting")
-
 	// Run servers
-	go func() { glog.Error(serverutil.ServeHTTPMetrics(*metricsAddr, serverutil.Readyz(sqldb))) }()
-	go serveHTTPGateway(ctx, lis, dopts, grpcServer,
-		pb.RegisterKeyTransparencyAdminHandlerFromEndpoint,
-	)
-	runSequencer(ctx, conn, directoryStorage)
+	g, gctx := errgroup.WithContext(ctx)
+	g.Go(func() error { return serverutil.ServeHTTPMetrics(*metricsAddr, serverutil.Readyz(sqldb)) })
+	g.Go(func() error {
+		return serverutil.ServeHTTPAPIAndGRPC(gctx, lis, *keyFile, *certFile,
+			grpcServer, conn, pb.RegisterKeyTransparencyAdminHandler)
+	})
+	g.Go(func() error { return runSequencer(gctx, conn, directoryStorage) })
 
-	// Shutdown.
-	glog.Errorf("Signer exiting")
+	glog.Errorf("Signer exiting: %v", g.Wait())
 }
 
-func runSequencer(ctx context.Context, conn *grpc.ClientConn,
-	directoryStorage dir.Storage) {
+func runSequencer(ctx context.Context, conn *grpc.ClientConn, directoryStorage dir.Storage) error {
```

Comment:
nit: This func never returns anything but nil - could drop the `error` return type and leave it out of the errgroup.

Reply:
Favoring consistency, I'll keep it :-)

Reply:
Probably helpful to throw a comment at the start documenting that it only logs errors that occur as part of running the sequencer then (to avoid anyone being misled by the signature).